src/jedi_bundle/bin/jedi_bundle_update_hash.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+# (C) Copyright 2022 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+# --------------------------------------------------------------------------------------------------
+
+import argparse
+from datetime import datetime as dt
+from jedi_bundle.utils.logger import Logger
+from jedi_bundle.utils.update_hash import update_hash
+
+
+def jedi_bundle_update_hash():
+
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('date', type=str,
+                        help='Provide a date using YYYY-MM-DD format to update pinned versions \n'
+                             'to the latest commit hashes before the provided date')
+
+    logger = Logger('JediBundleUpdateHash')
+    args = parser.parse_args()
+    date = args.date
+
+    # Convert date to datetime object
+    dt_object = dt.strptime(date, '%Y-%m-%d').astimezone()
```

Comment:
This is _probably_ fine, but it does make me a little bit nervous. The default behavior of `.astimezone()` is to convert everything to _local_ time, which means that the same code will produce subtly different times depending on exactly _where_ it runs.

If we want to have somewhat more predictable behavior, I might do something like this to always force midnight UTC time:

```python
from datetime import datetime, date, time, UTC
# [...]
dt_object = datetime.combine(date.fromisoformat(date), time(0, 0, tzinfo=UTC))
```